composer/utils/data.py

```diff
@@ -93,26 +98,60 @@ def pil_image_collate(batch: List[Tuple[Image.Image, Union[Image.Image, Tensor]]
     return image_tensor, target_tensor
 
 
-def add_dataset_transform(dataset, transform):
-    """Flexibly add a transform to the dataset's collection of transforms.
+def add_dataset_transform(dataset: VisionDataset,
+                          transform: Callable,
+                          is_tensor_transform: bool = False) -> torch.utils.data.Dataset:
+    """Add a transform to a dataset's collection of transforms. Inserts the transform
+    before or after torchvision.transforms.ToTensor(), or at the end of the collection of
+    ToTensor() is not present.
 
     Args:
-        dataset: A torchvision-like dataset
-        transform: Function to be added to the dataset's collection of transforms
+        dataset (torch.utils.data.Dataset): A torchvision-like dataset
+        transform (Callable): Function to be added to the dataset's collection of
+            transforms
+        is_tensor_transform (bool): Whether `transform` acts on data of the type
+        torch.Tensor. If `True`, and torchvision.transforms.ToTensor() is present in
+        `dataset`'s transforms, will insert the transform after ToTensor(). If `False` and
+        ToTensor() is present, will insert the transform before ToTensor(). If ToTensor()
+        is not present, the transform will be appended to the end of collection of
+        transforms. Default = `False`.
 
     Returns:
         The original dataset. The transform is added in-place.
     """
 
     if not hasattr(dataset, "transform"):
```

Comment:
Since the type annotation is a `VisionDataset`, this hasattr check is no longer needed.